### PR TITLE
Disable saml integration

### PR DESCRIPTION
**Pull request by Thanhphan1147**
Spec : ISD-185 (internal)

As authentication will be handled by the Matrix Authentication Service, the saml integration will need to be removed. Future revisions of synapse will therefore no longer support the saml integration.

Authentication via SAML is possible with Dex serving as a bridge ( via the dex-auth charm ) which will be possible once the oauth integration is added.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Manging Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by javierdelapuente — APPROVED** [review]
nice, just removing code!

File: src/matrix_auth_observer.py [hunk 1]
```diff
@@ -112,7 +112,6 @@ def _get_matrix_auth_provider_data(
         Returns:
             MatrixAuthConfiguration instance.
         """
-        # future refactor victim: this is repeated with saml
         homeserver = (
             charm_state.synapse_config.public_baseurl
```
Comment (amandahla) [inline src/matrix_auth_observer.py]: I think this could be removed too, right? Since it is done in charm_state.

Comment (amandahla) [reply src/matrix_auth_observer.py]: (I mean, the check)

Comment (Thanhphan1147) [reply src/matrix_auth_observer.py]: nice catch, I removed this logic.

**Review by amandahla — APPROVED** [review]

**Comment by amandahla** [comment]
Trivy vulnerability will be fixed here:
https://github.com/element-hq/synapse/pull/17985

**Pull request merged**